kernel/data-pipeline/core/src/main/java/org/apache/shardingsphere/data/pipeline/core/task/InventoryTask.java

```diff
@@ -122,10 +121,8 @@ public void onFailure(final Throwable throwable) {
     
     private PipelineChannel createChannel(final PipelineChannelCreator pipelineChannelCreator) {
         return pipelineChannelCreator.createPipelineChannel(1, records -> {
-            Record lastNormalRecord = RecordUtils.getLastNormalRecord(records);
-            if (null != lastNormalRecord) {
-                position.set(lastNormalRecord.getPosition());
-            }
+            Record lastRecord = records.get(records.size() - 1);
+            position.set(lastRecord.getPosition());
```

Comment:
If there's FinishedRecord, could we set it to position?

Looks `RecordUtils.getLastNormalRecord` could be kept to handle PlaceholderRecord, and also be updated to handle FinishedRecord?